### PR TITLE
Add support for ntlm authentication

### PR DESCRIPTION
**Pull request by samhanes**

### CODE REVIEW

File: tests/Paket.Tests/DependenciesFile/ParserSpecs.fs [hunk 1]
```diff
@@ -680,23 +680,38 @@ let ``should read config without versions``() =
     cfg.GetDependenciesInGroup(Constants.MainDependencyGroup).[PackageName "FAKE"].Range |> shouldEqual (VersionRange.AtLeast "0")
 
 
-let configWithPassword = """
-source http://www.nuget.org/api/v2 username: "tat� tata" password: "you got hacked!"
+let configWithPasswordNoAuthType = """
+source http://www.nuget.org/api/v2 username: "tat� tata" password: "you got hacked!"
```
Comment (matthid) [inline tests/Paket.Tests/DependenciesFile/ParserSpecs.fs]: something broke here

Comment (samhanes) [inline tests/Paket.Tests/DependenciesFile/ParserSpecs.fs]: Ooh yikes, I'll have a look!

**Comment by samhanes** [comment]
Yes, I think you're right - specifying NTLM in a file would mean it can't be used on non-windows.  But of course if your nuget feed requires windows auth, that's going to be a problem already.

I had considered limiting the auth type specification to config files for that reason, but I ultimately decided that if your nuget feed requires windows auth, you're already in trouble if you're not on windows.  Do you think that makes sense?

**Comment by matthid** [comment]
Yep, just asking for confirmation I guess :)

File: docs/content/nuget-dependencies.md [hunk 2]
```diff
@@ -36,14 +36,14 @@ source http://myserver/nuget/api/v2 // Custom feed.
...
-source http://example.com/nuget/api/v2 username: "user name" password: "the password"
+source http://example.com/nuget/api/v2 username: "user name" password: "the password" authtype: "basic"
 ```
 
 If you don't want to check your username and password into source control, you
 can use environment variables instead:
 
 ```paket
-source http://myserver/nuget/api/v2 username: "%PRIVATE_FEED_USER%" password: "%PRIVATE_FEED_PASS%"
+source http://myserver/nuget/api/v2 username: "%PRIVATE_FEED_USER%" password: "%PRIVATE_FEED_PASS%" authtype: "ntlm"
```
Comment (0x53A) [inline docs/content/nuget-dependencies.md]: You need either username/password OR ntlm, right?

Comment (0x53A) [inline docs/content/nuget-dependencies.md]: Or do you need to enter your windows account and password? It should be able to use the current windows account.

**Comment by 0x53A** [comment]
So i may be misunderstanding something, I haven't looked at this too deep to be honest.

But normally I understand WindowsAuthentication to mean that the current ambient user account will be used, and I DON'T need to enter my password anywhere.

Maybe this could help? https://stackoverflow.com/a/11414695/1872399

**Comment by samhanes** [comment]
@0x53A  Using the default credentials is certainly one use case - but in this case we'd like to use Windows credentials to a separate domain for authentication.  Locally we can store them in the windows credentials manager, but this still presents a problem on the build server.  The changes I've proposed would allow you to provide a username in the form of `domain\user` and then paket will those credentials to authenticate the request via NTLM.

File: docs/content/nuget-dependencies.md (hunk 2 above)
Comment (samhanes) [inline docs/content/nuget-dependencies.md]: No, the default credentials aren't used - just the authentication scheme.  See my reply to your other comment. :)

**Comment by 0x53A** [comment]
ok, makes sense.

I guess if someone wants default credentials they should implement it themselves ;D

**Comment by forki** [comment]
WOW! This looks great!

**Comment by samhanes** [comment]
Thanks - you guys move fast!

**Pull request merged**